### PR TITLE
Fix test that fails if DEV=True in local env

### PR DESCRIPTION
**Pull request by stevejalim**
This changeset covers an edge case where, if you have DEV=True in a local .env, the TestRelayFxAButton.test_relay_fxa_button test would fail.

This is because, despite DEV being patched to False, RELAY_PRODUCT_URL was set to something else before the patching. So, adding an additional override was needed to keep the test stably passing

### CODE REVIEW

File: bedrock/mozorg/tests/test_helper_misc.py [hunk 1]
```diff
@@ -1101,6 +1101,7 @@ def test_pocket_no_redirect_adjust_url(self):
 
 
 @override_settings(FXA_ENDPOINT=TEST_FXA_ENDPOINT)
+@override_settings(RELAY_PRODUCT_URL="https://relay.firefox.com/")
 @override_settings(DEV=False)
```
Comment (alexgibson) [inline bedrock/mozorg/tests/test_helper_misc.py:1105]: I don't think this `Dev=False` line is needed, right?

**Review by alexgibson — APPROVED** [review]
r+wc
